src/storage/src/hummock/compactor.rs

```diff
@@ -865,24 +902,61 @@ impl Compactor {
             filter_key_extractor_manager,
             memory_limiter,
             sstable_id_manager,
+            task_progress: task_progress.clone(),
         });
         let (shutdown_tx, mut shutdown_rx) = tokio::sync::oneshot::channel();
         let stream_retry_interval = Duration::from_secs(60);
+        let task_progress_update_interval = Duration::from_millis(1000);
         let join_handle = tokio::spawn(async move {
-            let process_task = |task, compactor_context, sstable_store, hummock_meta_client| async {
-                match task {
-                    Task::CompactTask(compact_task) => {
-                        Compactor::compact(compactor_context, compact_task).await;
-                    }
-                    Task::VacuumTask(vacuum_task) => {
-                        Vacuum::vacuum(vacuum_task, sstable_store, hummock_meta_client).await;
-                    }
-                    Task::FullScanTask(full_scan_task) => {
-                        Vacuum::full_scan(full_scan_task, sstable_store, hummock_meta_client).await;
+            let process_task =
+                |task,
+                 compactor_context,
+                 sstable_store,
+                 hummock_meta_client,
+                 shutdown_map: CompactionShutdowSenderMap| async move {
+                    match task {
+                        Task::CompactTask(compact_task) => {
+                            let (tx, rx) = oneshot_channel();
+                            shutdown_map
+                                .lock()
+                                .unwrap()
+                                .insert(compact_task.task_id, tx);
+                            Compactor::compact(compactor_context, compact_task, rx).await;
+                        }
+                        Task::VacuumTask(vacuum_task) => {
+                            Vacuum::vacuum(vacuum_task, sstable_store, hummock_meta_client).await;
+                        }
+                        Task::FullScanTask(full_scan_task) => {
+                            Vacuum::full_scan(full_scan_task, sstable_store, hummock_meta_client)
+                                .await;
+                        }
+                        Task::CancelCompactTask(cancel_compact_task) => {
+                            if let Some(tx) = shutdown_map
+                                .lock()
+                                .unwrap()
+                                .remove(&cancel_compact_task.task_id)
+                            {
+                                tracing::warn!(
```

Comment:
Maybe this trace is not necessary. The receiver will already report task cancelled externally.